index.js

```diff
@@ -1,84 +1,102 @@
-const SlackClient = require('@slack/client').WebClient;
+const express = require("express");
+const http = require("http");
+const bodyParser = require("body-parser");
+const slackInteractiveMessages = require("@slack/interactive-messages");
+const { createSlackEventAdapter } = require("@slack/events-api");
+const SlackClient = require("@slack/client").WebClient;
 
-const  { initiationMessage, categoryMessage, witnessesMessage, getConfirmationMessage } = require('./modules/messages'); 
-const actions = require('./modules/actions');
+const { isDateValid } = require("./modules/regex");
 
-const dotenv = require('dotenv');
+const { isDateFuture } = require("./modules/future_date_validation");
+
+const {
+  initiationMessage,
+  categoryMessage,
+  witnessesMessage,
+  getConfirmationMessage
+} = require("./modules/messages");
+const actions = require("./modules/actions");
+
+const dotenv = require("dotenv");
 
 dotenv.load();
 
-const slackEvents = createSlackEventAdapter(process.env.SLACK_VERIFICATION_TOKEN);
+const slackEvents = createSlackEventAdapter(
+  process.env.SLACK_VERIFICATION_TOKEN
+);
 const sc = new SlackClient(process.env.SLACK_BOT_TOKEN);
-const slackMessages = slackInteractiveMessages
-  .createMessageAdapter(process.env.SLACK_VERIFICATION_TOKEN);
+const slackMessages = slackInteractiveMessages.createMessageAdapter(
+  process.env.SLACK_VERIFICATION_TOKEN
+);
 const port = process.env.PORT || 3000;
 
 const confirmIncident = (user, channel) => {
   const data = actions.tempIncidents[user];
-  sc.chat.postMessage(
-    channel,
-    '',
-    getConfirmationMessage(data),
-  (err, res) => {
+  sc.chat.postMessage(channel, "", getConfirmationMessage(data), (err, res) => {
     // console.log(res);
   });
-}
+};
 
-slackEvents.on('message', (event) => {
+slackEvents.on("message", event => {
   // do not respond to edited messages or messages from me
-  if (event.subtype === 'bot_message' || event.subtype === 'message_changed') {
+  if (event.subtype === "bot_message" || event.subtype === "message_changed") {
     return;
-  };
+  }
 
   const userId = event.user;
   if (!actions.tempIncidents[userId]) {
-
-    sc.chat.postMessage(
-      event.channel,
-      '',
-      initiationMessage,
-    (err, res) => {
+    sc.chat.postMessage(event.channel, "", initiationMessage, (err, res) => {
       // console.log(res);
     });
-
   } else {
-
     const currentStep = actions.tempIncidents[userId].step;
-    switch(currentStep) {
+    switch (currentStep) {
       case 0:
+        if (isDateValid(event.text) === false) {
+          sc.chat.postMessage(
+            event.channel,
+            "Non-existent date provided. Try again.",
+            (err, res) => {
+              // console.log(res);
+            }
+          );
+
+          break;
+        }
+
+        if (isDateFuture(event.text) === true) {
+          sc.chat.postMessage(
+            event.channel,
+            "Non-existent date provided. Try again.",
```

Comment:
give a more descriptive message, i.e. You cannot report a future incident or Invalid date entered (dd-mm-yy)

Reply:
Cool